creator-node/src/snapbackSM/secondarySyncHealthTracker.js

```diff
@@ -13,97 +13,13 @@ const Outcomes = Object.freeze({
-    await this._recordSyncRequestOutcome(secondary, wallet, syncType, true)
-  },
-
-  async recordFailure (secondary, wallet, syncType) {
-    await this._recordSyncRequestOutcome(secondary, wallet, syncType, false)
-  },
-
-  async _recordSyncRequestOutcome (secondary, wallet, syncType, success = true) {
-    try {
-      const outcome = success ? Outcomes.SUCCESS : Outcomes.FAILURE
-      const redisKey = this._getRedisKeyPattern(secondary, wallet, syncType, outcome)
-
-      // incr() will create key with value 0 if non-existent
-      await redisClient.incr(redisKey)
-
-      // Set key expiration time (sec) in case it hasn't already been set (prob not most efficient)
-      await redisClient.expire(redisKey, DailyRedisKeyExpirationSec)
-
-      logger.info(`SecondarySyncHealthTracker:_recordSyncRequestOutcome || Recorded ${redisKey}`)
-
-      // Swallow error + log
-    } catch (e) {
-      logger.error(`SecondarySyncHealthTracker:_recordSyncRequestOutcome Error || ${e.message}`)
-    }
-  },
-
-  /**
-   * Get SyncRequest outcome metrics for all secondaries, users, and days
-   */
-  async getAllSyncMetrics () {
-    try {
-      const pattern = this._getRedisKeyPattern('*', '*', '*', '*', '*')
-
-      return this._getMetricsMatchingPattern(pattern)
-    } catch (e) {
-      logger.error(`secondarySyncHealthTracker - getAllSyncMetrics() Error || ${e.message}`)
-      return {}
-    }
-  },
-
-  /**
-   * Get SyncRequest outcome metrics for given secondary, all users, and all days
-   */
-  async getSyncMetricsForSecondary (secondary) {
-    try {
-      const pattern = this._getRedisKeyPattern(secondary, '*', '*', '*', '*')
-
-      return this._getMetricsMatchingPattern(pattern)
-    } catch (e) {
-      logger.error(`secondarySyncHealthTracker - getSyncMetricsForSecondary() Error || ${e.message}`)
-      return {}
-    }
-  },
-
-  /**
-   * Get SyncRequest outcome metrics for given secondary, given user, and all days
-   */
-  async getSecondarySyncMetricsForSecondaryForUser (secondary, wallet) {
-    try {
-      const pattern = this._getRedisKeyPattern(secondary, wallet, '*', '*', '*')
-
-      return this._getMetricsMatchingPattern(pattern)
-    } catch (e) {
-      logger.error(`secondarySyncHealthTracker - getSecondarySyncMetricsForUser() Error || ${e.message}`)
-      return {}
-    }
-  },
-
-  /**
-   * Get SyncRequest outcome metrics for given secondary, given user, and for today
-   */
-  async getSecondarySyncMetricsForSecondaryForUserForToday (secondary, wallet) {
-    try {
-      const today = new Date().toISOString().split('T')[0] // format: YYYY-MM-DD
-      const pattern = this._getRedisKeyPattern(secondary, wallet, '*', '*', today)
-
-      return this._getMetricsMatchingPattern(pattern)
-    } catch (e) {
-      logger.error(`secondarySyncHealthTracker - getSecondarySyncMetricsForUserForToday() Error || ${e.message}`)
-      return {}
-    }
-  },
-
+const Utils = {
```

Comment:
im not particularly familiar with this code structure convention. what was the benefit of grouping functions into consts, and then reassigning them to `SecondarySyncHealthTracker` when exported?

Reply:
just for logical separation, helps me understand the code separation a lot better. trying to avoid the situation in other classes where we have 20-30 functions all at the same level even though there is a clear hierarchical structure
makes no difference logically

Reply:
I understand how this format possibly helps with clearly defining the hierarchy. however, i have a few points against this: 

1. it's a little strange that in these conts, you have inner functions referencing the consts itself (e.x: in `Utils` const, the fn `_getMetricsMatchingPattern()` calls `Utils._getAllKeysMatchingPattern(pattern)`). looks like we could've just made this a static class and/or keep the prior commit version of this file
2. it's redundant to have outer consts like `Setters` and `Getters` to group the actual fns. getters/setters should be very clearly obvious via the function name and grouped structurally in the same area of the file. 
3. this file introduces a new pattern that isn't around in the code base. I rather keep to current structures we have in the code base (no hierarchal structuring of fns, just arranging fns in the proper place and possibly introducing strong comments like `// ========= GETTERS ========= ` to divide the file)